backend/inspirehep/hal/utils.py

```diff
@@ -0,0 +1,145 @@
+
+from itertools import chain
+
+from flask import current_app
+from inspire_dojson.utils import get_recid_from_ref
+from inspire_utils.name import ParsedName
+from inspire_utils.record import get_value
+
+from inspirehep.records.api import InspireRecord
+
+
+def get_authors(record):
+    """Return the authors of a record.
+    Queries the Institution records linked from the authors affiliations
+    to add, whenever it exists, the HAL identifier of the institution to
+    the affiliation.
+    Args:
+        record(InspireRecord): a record.
+    Returns:
+        list(dict): the authors of the record.
+    Examples:
+        >>> record = {
+        ...     'authors': [
+        ...         'affiliations': [
+        ...             {
+        ...                 'record': {
+        ...                     '$ref': 'http://localhost:5000/api/institutions/902725',
+        ...                 }
+        ...             },
+        ...         ],
+        ...     ],
+        ... }
+        >>> authors = get_authors(record)
+        >>> authors[0]['hal_id']
+        '300037'
+    """
+    hal_id_map = _get_hal_id_map(record)
+
+    result = []
+
+    for author in record.get("authors", []):
+        affiliations = []
+
+        parsed_name = ParsedName.loads(author["full_name"])
+        first_name, last_name = parsed_name.first, parsed_name.last
+
+        for affiliation in author.get("affiliations", []):
+            recid = get_recid_from_ref(affiliation.get("record"))
+            if recid in hal_id_map and hal_id_map[recid]:
+                affiliations.append({"hal_id": hal_id_map[recid]})
+
+        result.append(
+            {
+                "affiliations": affiliations,
+                "first_name": first_name,
+                "last_name": last_name,
+            }
+        )
+
+    return result
+
+
+def get_conference_record(record, default=None):
+    """Return the first Conference record associated with a record.
+    Queries the database to fetch the first Conference record referenced
+    in the ``publication_info`` of the record.
+    Args:
+        record(InspireRecord): a record.
+        default: value to be returned if no conference record present/found
+    Returns:
+        InspireRecord: the first Conference record associated with the record.
+    Examples:
+        >>> record = {
+        ...     'publication_info': [
+        ...         {
+        ...             'conference_record': {
+        ...                 '$ref': '/api/conferences/972464',
+        ...             },
+        ...         },
+        ...     ],
+        ... }
+        >>> conference_record = get_conference_record(record)
+        >>> conference_record['control_number']
+        972464
+    """
+    pub_info = get_value(record, "publication_info.conference_record[0]")
+    if not pub_info:
+        return default
+
+    return InspireRecord.get_record_by_pid_value(get_recid_from_ref(pub_info), "con")
+
+
+def get_divulgation(record):
```

Comment:
Returning 1/0 instead True/False this is is so "C" way, but it's not our API so we can't do anything with this... :/